tests/integration/components/denali-select-test.js

```diff
@@ -0,0 +1,145 @@
+
+    assert.dom('select').exists('DenaliSelect renders a select element.');
+
+    assert.dom('option').hasText('Item 1', 'DenaliSelect option exists and renders an object.');
+  });
+
+  test('an option is active', async function (assert) {
+    assert.expect(1);
+
+    await render(hbs`
+      <DenaliSelect @options={{this.options}} @selectedOption={{this.selectedOption}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    this.set('options', [option]);
+    this.set('selectedOption', option);
+    assert.equal(find('select').selectedIndex, 0);
+  });
+
+  test('an option is disabled', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`
+      <DenaliSelect @options={{this.options}} @disabledOptions={{this.disabled}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    this.set('options', [option]);
+    assert.dom('option').doesNotHaveAttribute('disabled', 'DenaliSelect options are not disabled by default');
+
+    this.set('disabled', [option]);
+    assert.dom('option').isDisabled('DenaliSelect option is disabled when `@disabledOptions` arg is set');
+  });
+
+  test('it supports small size', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`
+      <DenaliSelect @options={{this.options}} @isSmall={{this.isSmall}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    this.set('options', [option]);
+    assert.dom('div').doesNotHaveClass('is-small', 'DenaliSelect does not have small styling by default');
+
+    this.set('isSmall', 'true');
+    assert.dom('div').hasClass('is-small', 'DenaliSelect has a small size when `@isSmall` arg is set to true');
+  });
+
+  test('it supports inverse colors', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`
+      <DenaliSelect @options={{this.options}} @isInverse={{this.isInverse}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    this.set('options', [option]);
+    assert.dom('div').doesNotHaveClass('is-inverse', 'DenaliSelect does not have inverse styling by default');
+
+    this.set('isInverse', 'true');
+    assert.dom('div').hasClass('is-inverse', 'DenaliSelect has inverse styling when `@isInverse` arg is set to true');
+  });
+
+  test('it handles change events', async function (assert) {
+    assert.expect(2);
+
+    await render(hbs`
+      <DenaliSelect
+        @options={{this.options}}
+        @selectedOption={{this.selectedOption}}
+        @disabledOptions={{this.disabled}}
+        @onChange={{this.onChange}} as |option|>
+        {{option.text}}
+      </DenaliSelect>
+    `);
+
+    this.set('options', [{ text: 'Item 1' }, { text: 'Item 2' }, { text: 'Item 3' }]);
+    this.set('selectedOption', this.options[2]);
+    assert.equal(find('select').selectedIndex, 2);
+
+    this.set('onChange', (option) => {
+      this.set('selectedOption', option);
```

Comment:
assert.equal(option, this.options[1], 'DenaliSelect `@onChange` action provides the object selected')